tests/unit/s2n_tls13_pq_handshake_test.c

```diff
@@ -769,7 +796,17 @@ int main()
                 .expected_curve = ec_if_no_mlkem,
                 .hrr_expected = false,
                 .len_prefix_expected = false,
-        }
+        },
+
+        /* Confirm that pure MLKEM1024 is negotiable; fall back to EC when MLKEM is not supported. */
+        {
+                .client_policy = &pure_mlkem1024_test_policy,
+                .server_policy = &pure_mlkem1024_test_policy,
+                .expected_kem_group = null_if_no_pure_mlkem_1024,
+                .expected_curve = ec_if_no_mlkem,
+                .hrr_expected = false,
```

Comment:
Curious if you think we need more test coverage over pure pq in the Hello Retry scenario. If the client doesn't send a pure pq keyshare, but does support it, can the server do a HRR and get the correct kem group?